### ISSUE TITLE
Disable "inteligent" export type

### ISSUE THREAD

**Issue by SteveDaulton** [audacity/audacity#5402]
### Your idea

In 3.4 beta, if the project contains multiple tracks or any labels, "Export Multiple" is automatically selected.
I'm proposing that  this behaviour becomes optional.

### Problem to be solved

Almost all of my projects have either more than one track, or one or more labels. In almost all cases I want to export a mix of all non-muted tracks. I suspect that this is a common workflow used by many thousands of Audacity users.

With the new "intelligent" export type selection, I must remember to reset the export type to "Entire Project" every time I export, or the export will be wrong.

There's also a potential "gotcha":
Labels are frequently used for labelling parts of a project. The new behaviour has the potential for unwary users to lose work. For example:
1. Import a file
2. Edit the project and during the process add a region label to mark a region of interest
3. "Ctrl + Shift +  E"
4. Check the destination and file name
5. Click  "Export" -> "OK"
6. Close  the project without saving.
Unless you were paying close attention in step 4, only the labeled region will be exported and all of your editing work is lost.

### Prior art

All previous versions of Audacity used Export Multiple only when explicitly selected by the user.

### Additional context

The new behaviour is likely to be a nice convenience to a subset of Audacity users. Without a large scale poll it is impossible to know how big or small that subset is, but I suspect that it is a relatively small minority of users. Because of it's usefulness I agree that the new feature should be retained, but for the (possibly millions of ) users that have no use for the feature, it would be useful to be able to turn it off.

**Comment by DavidBailes** [comment]
> In 3.4 beta, if the project contains multiple tracks or any labels, "Export Multiple" is automatically selected.
> I'm proposing that this behaviour becomes optional.

@SteveDaulton . Thanks for highlighting this behaviour. As far as I can see, "Export multiple" is automatically selected only if the project contains one or more labels, and not if it just contains multiple audio tracks.

As Steve has already said, I think that this behaviour is lightly to cause users to loose their work. Label tracks have many uses, not just the one of specifying how audio is split into a number of files. So a user could easily not notice that the Export range has been silently changed to "multiple files". This is particularly the case for users of screen readers.

@LWinterberg @Tantacrul . Please could this issue be reviewed before 3.4.0 is released?

**Comment by petersampsonaudacity** [comment]
I totallt support @SteveDaulton 's and @DavidBailes 's points

**Comment by SteveDaulton** [comment]
> As far as I can see, "Export multiple" is automatically selected only if the project contains one or more labels, and not if it just contains multiple audio tracks.

With Audacity 3.4.0 beta (commit b3aed4 from the 3.4 release branch), "Multiple" is automatically selected if the project contains labels and/or multiple tracks. So just having a voice track and a background music track will automatically select export multiple.

**Comment by DavidBailes** [comment]
> > As far as I can see, "Export multiple" is automatically selected only if the project contains one or more labels, and not if it just contains multiple audio tracks.
> 
> With Audacity 3.4.0 beta (commit b3aed4 from the 3.4 release branch), "Multiple" is automatically selected if the project contains labels and/or multiple tracks. So just having a voice track and a background music track will automatically select export multiple.

I don't think so. Note that the group of export radio buttons remembers the last setting.
So what happens if you:
1. Open audacity, create an audio track, export the entire project, close Audacity.
2. Open audacity again, create two audio tracks, open the export dialog. Is it set to export multiple?

The relevant code in the dialog's constructor is:
```
   if(!hasLabels)
   {
      mSplitByLabels->Disable();
      mSplitByTracks->SetValue(true);
      if(ExportAudioExportRange.Read() != "split")
         mSplitsPanel->Hide();
   }
   else
   {
      mRangeSplit->SetValue(true);
      mSplitByLabels->SetValue(true);
   }
```
where mRangeSplit is the "multiple files" radio button, and it's the only place in the code where it's set.

**Comment by DavidBailes** [comment]
Note that I've since significantly edited my last comment.

**Comment by petersampsonaudacity** [comment]
>With Audacity 3.4.0 beta (commit b3aed4 from the 3.4 release branch), "Multiple" is automatically selected if the project contains labels and/or multiple tracks. So just having a voice track and a background music track will automatically select export multiple.

That is **_NOT_** the case when I test on W10 with the latest 3.4.0 Beta.

Multiple Export (by labels) is only defaulted in the New Export dialog if and only if at least one label track is present.

But I think this is poor UX as labels and label tracks can be used for may more things than simply just defining songs within a track.  In face some of the Audacity Analyzers _(and most/all of the QM Vamp analyzers)_ create label tracks for their output.

**Comment by DavidBailes** [comment]
> a) IF the user has an audio track (or multiple audio tracks) and an empty label track, THEN export multiple via **tracks** is default offered

@petersampsonaudacity . I don't think that is correct. See my description of the behaviour in my previous comment: https://github.com/audacity/audacity/issues/5402#issuecomment-1773194153
In particular, that if there are no labels, then the previous export range is selected.

**Comment by petersampsonaudacity** [comment]
Yes @DavidBailes 

I think I see what you mean, I did

1. get 2 audio tracks
2. File >Export Audio
3. Observe: multiple export via tracks is offered
4. make a selection in one (or both) track(s)
5. Observe: "Current Selection" is offerd as the range
6. remove the

**Comment by DavidBailes** [comment]
> For my money it would be much simpler and more understandable if it always defaulted to offering **Entire Project** as the **Export Range**.
> 
> Would that suit your accessibility criteria David?

@petersampsonaudacity Are you suggesting that the initial default is Entire project, and that both it doesn't remember the previous setting, and it doesn't automatically set the export range to multiple files if there are one or more labels in  the project?

From an accessibility usability for keyboard users point of view, that would be fine. The current behaviour is not:
1. Given that labels have a number of uses, silently changing the export range (which is towards the end of the dialog) to multiple files is very unhelpful.
2. Setting the initial export range to the previous value is less of a problem than the first issue, as the user has previously actively changed the setting. However they to have to remember that they've done this, and this could have been some time ago. Again, having the export range radio buttons towards the end of the dialog rather than much earlier doesn't help with this.

**Comment by petersampsonaudacity** [comment]
> Are you suggesting that the initial default is Entire project, and that both it doesn't remember the previous setting, and it doesn't automatically set the export range to multiple files if there are one or more labels in the project?
>
>From an accessibility usability for keyboard users point of view, that would be fine. The current behaviour is not

@DavidBailes   yepper, that's **_exactly_** what I am suggesting.

Simple, straightforward _(no context dependency as you would get with a "use last setting" which may not be appropriate on next use)_.

**Comment by LWinterberg** [comment]
I don't think things should ever default to something other than "Entire Project".

**Comment by petersampsonaudacity** [comment]
>I don't think things should ever default to something other than "Entire Project".

@LWinterberg  glad to hear that Leo - and in which case any chance we can get this fixed in time for 3.4.0 please?

As:
a) it mars the otherwise lovely new export dialog that we have for 3.4.0,
b) makes life harder in terms of accessibility as @DavidBailes has explained.

**Comment by crsib** [comment]
Yep, will fix early next week

**Comment by petersampsonaudacity** [comment]
> Yep, will fix early next week

Thanks Dmitry

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @crsib 's branch build:  audacity-win-3.4.0-alpha-20231030+c5fb3b0-x64

This works on first use if you have
a) multiple tracks
b) a non-empty label track
c) a selection
Then regardless if the above the "Entire Project" is properly offered 

BUT on next use it still remembers the previously used export range - and sometime inappropriately so - try these steps

1. purge the Audacity settings folder
2. get some audio - single track
3. add some labels
4. File > Export Audio
5. Observe: "Entire Project" is the default range offered
6. set the rang to "Multiple Files"
7. Export button to export the labeled ranges
8. Delete the Label Track
9. File > Export Audio
10. Observe: "Multiple Files" is still the default range offered (even though there a no labels and only a single track)

It's a bit smarter with Export - but not totally smart
1. purge the Audacity settings folder
2. get some audio - single track
3. select a portion of the track
4. File > Export Audio
5. Observe: "Entire Project" is the default range offered
6. set the range to be "Current selection"
7. Export button to export the selection
8. make a different selection
9. File > Export Audio
10. Observe: "Current selection" is the default range offered (remembered from last use)
11. Export button to export the selection
12. make a point "selection" - i.e. set a cursor position
13.  File Export >Audio
14. Observe: "Entire Project" is offered as the default range

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @crsib 's latest updated branch build for this:  audacity-win-3.4.0-alpha-20231030+6cc55d0-x64

This now looks **_much_** better to me.

Yes it now remembers that previously used range setting but now only applies that if and when it is appropriate - if it is not appropriate then "Entire Project" is offered.

@LWinterberg @crsib @vsverchinsky This looks good to go to me

**Comment by chinakov** [comment]
Tested!
